convert_hf_to_gguf.py

```diff
@@ -2145,6 +2145,9 @@ def set_vocab(self):
                 toktype = SentencePieceTokenTypes.UNUSED
             elif tokenizer.IsByte(token_id):
                 toktype = SentencePieceTokenTypes.BYTE
+            # take care of ununsed raw token
+            if piece.startswith('[UNUSED'):
+                toktype = SentencePieceTokenTypes.UNKNOWN
```

Comment:
I wonder, does it work without this if the checks for `UNKNOWN` below are replaced with checks for `UNUSED`?

Reply:
hi, these raw tokens are in fact used and updated by  [added_tokens_decoder](https://huggingface.co/internlm/internlm2_5-7b-chat/blob/6a4dae231b8d01fe13736e334589d0d864d1ea23/tokenizer_config.json#L4). And there are asserations in line [2180](https://github.com/ggerganov/llama.cpp/blob/3d04d337caf4eef74b4b6d15b995161132811218/convert_hf_to_gguf.py#L2180) and 2199 which is aligned with what's done in [phi3](https://github.com/ggerganov/llama.cpp/blob/a59f8fdc85e1119d470d8766e29617962549d993/convert_hf_to_gguf.py#L1921). The tokens are all read as `SentencePieceTokenTypes.NORMAL` and the assertations would fail if remove or change to `UNUSED` type.